sdks/go/pkg/beam/core/metrics/metrics.go

```diff
@@ -0,0 +1,447 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package metrics handles Beam metrics.
+//
+// Metrics in the Beam model are uniquely identified by a namespace, a name,
```

Comment:
I couldn't find one on the beam site that wasn't JavaDoc or PyDoc. Ideally we'd refer to one of the design docs, or a spec, but such things don't exist as far as I could find. One will probably turn up, but I don't believe it's worth blocking this PR over.

Reply:
Done, I finally found  http://s.apache.org/beam-metrics-api , which is now linked in the package line.

It's an older doc, so we aren't 1:1 with it (eg we don't permit metric queries at this time), but if there's a fresher one, I couldn't find it. The upshot is that I now use the term metric cells instead of metric containers.